src/codeu/chat/client/commandline/Chat.java

```diff
@@ -47,9 +46,22 @@ public Chat(Context context) {
   //
   public boolean handleCommand(String line) {
 
-    final Scanner tokens = new Scanner(line.trim());
+    final List<String> args = new ArrayList<>();
 
-    final String command = tokens.hasNext() ? tokens.next() : "";
+    final Tokenizer tokenizer = new Tokenizer(line);
+
+    //tokenizing the input, and adding each token to the ArrayList
+    try {
+      for (String token = tokenizer.next(); token != null; token = tokenizer.next()) {
```

Comment:
It seems like it would be useful if Tokenizer implemented `Iterator<String>`, that way you could just do `tokenizer.forEachRemaining(args::add);`